### PR TITLE
#48 PyPI publishing

### PR DESCRIPTION
**Pull request by absurdfarce**

### CODE REVIEW

File: python/adelphi/setup.py
Comment (absurdfarce) [inline python/adelphi/setup.py:44]: python-driver includes pypy support here.  We're _probably_ fine on pypy but I'm not explicitly testing on it so leave it off for now

File: python/adelphi/setup.py
Comment (absurdfarce) [inline python/adelphi/setup.py:45]: python-driver includes pypy support here.  We _should_ be fine on pypy but I haven't tested it explicitly (and didn't want to introduce that as an ongoing thing) so for now I just dropped it.

File: python/adelphi/setup.py
Comment (absurdfarce) [inline python/adelphi/setup.py:30]: python-driver includes a "Documentation" link here which presumably shows up in the "Project links/Documentation" section on PyPI.  This documentation would presumably be specific to the Python package so we could make this a link to the README on Github... not sure what else would work there.

**Comment by absurdfarce** [comment]
@grighetto and @jdonenine : Note that this PR is branched off of #71  rather than master.  I kinda doubted there would be any overlap between the two but just in case I decided to work off the more recent feature branch.  This PR will be rebased once #71 is merged to master so please don't merge this guy until we reach that point!

File: python/adelphi/README.md
Comment (jdonenine) [inline python/adelphi/README.md:2]: What do you think about adding a quick intro block to explain the high level of the script and what all it can do?  That will give a bit more context to the detailed commands you cover below, I think.

Comment (jdonenine) [reply python/adelphi/README.md:2]: Maybe it can include some info on how to install the package via pip and what the basic system requirements are (python version, anything else)?

File: python/adelphi/README.md
Comment (jdonenine) [inline python/adelphi/README.md:6]: If we change the output structure so that we get a directory and a series of files plus the metadata file we'll need to remember to come back here and update this.

Comment (jdonenine) [reply python/adelphi/README.md:6]: Based on the help dump below, I think this command is optional, and if it's left out you get all keyspaces by default - can we include a note about that here?

Comment (jdonenine) [reply python/adelphi/README.md:6]: Actually, I guess looking at the other blocks here that default behavior is more for the --keyspaces option than it is the individual commands, so no need to repeat it over and over again.  But, it makes me wonder if we shouldn't pull the options section up higher and then expand on the detailed commands afterwards maybe?

File: python/adelphi/README.md
Comment (jdonenine) [inline python/adelphi/README.md:67]: Here I don't think that noting the default as True makes it clear which is the default since it's sort of an either/or situation, not a single flag with a boolean variable.

File: python/adelphi/setup.py
Comment (jdonenine) [reply python/adelphi/setup.py:30]: If we can link back to our docs in the readme that seems like the best scenario so that we only have to manage it in one place over time.

**Review by jdonenine — CHANGES_REQUESTED** [review]

File: python/adelphi/README.md
Comment (grighetto) [inline python/adelphi/README.md:14]: `keyspaces` has no `=` and output-dir` has it, I think that could be confusing. I tend do prefer having the equals sign, but I don't care much either way as long we're consistent.

File: python/adelphi/README.md
Comment (grighetto) [reply python/adelphi/README.md:6]: typo: necesary

File: python/adelphi/README.md
Comment (grighetto) [inline python/adelphi/README.md:28]: maybe we can call out here that the `schemas will be *publicly* available in anonymized form` (or something along those lines), such that no one has any surprises

File: python/adelphi/README.md
Comment (grighetto) [inline python/adelphi/README.md:90]: that `click` lib makes this a little weird, sometimes options come before, other times after. I understand it gives some positional context to the args, but I'd prefer to pass the `--aaa`  options at whatever position and it would still handle it. Not related to this PR of course, but we can probably check later if we can be more flexible here.

File: python/adelphi/setup.py
Comment (absurdfarce) [reply python/adelphi/setup.py:30]: I wondered about that @jdonenine, but I held off since we were already linking to lots of stuff on Github.  cassandra-driver links to some DataStax-managed docs but I think that might just be a historical accident; I'm not sure that's the model we should aim for going forward.

I'll make this a link to the README for now, although weirdly it'll be a link to the README being added by this very pull request.  Remember that the docs we link to here should be specific to the Python package in question so we don't want to link to https://github.com/datastax/adelphi/blob/master/README.md.... I think we'd want https://github.com/datastax/adelphi/blob/master/python/adelphi/README.md instead.

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:2]: Yeah, I think that makes sense @jdonenine.  It seemed weird to add that kind of thing to a doc that's appearing on PyPI where presumably users are either already aware of those details of they're specified via classifiers and other metadata in setup.py.  But the README here is serving two distinct roles; it's providing content for PyPI _and_ it's serving as a general README in the Github repo.  The block you're describing makes a ton of sense for the second role so I'll add that in there.

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:6]: Your comments did highlight a couple things that are missing here @jdonenine :

* If keyspaces aren't specified all keyspaces are selected, at least initially
* System keyspaces are removed from the list whether you explicitly specify keyspaces or leave them out and select everything

I'll add comments about both those points to the docs.

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:67]: Fair point.  This is a weirdness of how the Python CLI arg lib we're using handles things.  When you declare a boolean flag it adds two CLI args, one for the name and one as "no-" name.  The flag is then turned on or off based on which one you add, so there is something of a decoupling there.

I think I'll change the default to something like "anonymization is enabled" to make that clearer.

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:14]: Good catch @grighetto , will fix in the next push

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:90]: I see your point... kinda.  I personally like the fact that the position of the args is used to be very explicit about what is command-specific and what is just used as part of running the app.

I _believe_ click supports the ability for command declarations to "inherit" options from their parent container (or something along those lines... I forget exactly how it works now).  Upshot is that all the args which are currently not tied to a command would devolve into command-specific args, which I believe would accomplish what you're describing.  I guess I'm not completely opposed to that; I even experimented with something like that when click first was brought in.  I personally still like the clear demarcation of arg types but I readily agree this may be an idiosyncrasy about me. :)

File: python/adelphi/setup.py
Comment (jdonenine) [reply python/adelphi/setup.py:30]: 👍  on it needing to be the python script specific README, not the general repo one.

File: python/adelphi/README.md
Comment (jdonenine) [reply python/adelphi/README.md:2]: Agree on people getting here likely by knowing what they were looking for, so not a lot of detail necessary I'd think.

Comment (absurdfarce) [reply python/adelphi/README.md:2]: I tried to address this in the "Introduction" section.  Take a look and see if you're satisfied with what's there @jdonenine

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md]: I wound up adding a section to the "Options" block describing how keyspace selection works.  I think it covers all the high points we discussed here.  Can you guys double-check and confirm that everything that needs to be there is there and that it's clear?

Comment (absurdfarce) [reply python/adelphi/README.md]: I wound up adding a section to the "Options" block describing how keyspace selection works.  I think it covers all the high points we discussed here.  Can you guys double-check and confirm that everything that needs to be there is there and that it's clear?

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md]: This should be fixed now; I added an explicit "=" for all examples in the docs

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:47]: This concept made it's way into the "Introduction"; it seemed important enough to warrant an explanation right up front.  Please take a look and let me know if that addresses the concern here.

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:86]: Okay, this now looks like...

```
...
  --anonymize / --no-anonymize  Enable/disable schema anonymization  [default:
                                anonymization enabled]
...
```

To me that's clearer all around.  Whaddya think @jdonenine ?

File: python/adelphi/README.md
Comment (grighetto) [inline python/adelphi/README.md:123]: typo: paramters

File: python/adelphi/README.md
Comment (grighetto) [inline python/adelphi/README.md:12]: I'd highlight the last part about visibility with some code-block style or at least move it to a new paragraph.

**Review by grighetto — APPROVED** [review]
Left a minor suggestion and found a typo, otherwise 👍

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md]: Good find, fixed!

File: python/adelphi/README.md
Comment (absurdfarce) [reply python/adelphi/README.md:12]: Hmmm, the content of that line fits in with the rest of that paragraph (in fact that paragraph was structured so that it would lead up to that statement) so I don't really want to decouple them.  But I think perhaps making just that sentence bold would satisfy both of us.  I'll give that a shot.

**Comment by absurdfarce** [comment]
I fixed the typo you identified @grighetto and was able to come up with a way to emphasize the public nature of the schema repo that I think will make both of us happy.  Would you mind taking a quick look to confirm that you're good?

File: python/adelphi/README.md
Comment (jdonenine) [reply python/adelphi/README.md:2]: I like it, thanks for adding.

**Review by jdonenine — APPROVED** [review]
Good from my perspective, I'll let @grighetto wrap up on the couple of comments he originally had and finish things up.  Thanks for adding the docs!

**Review by grighetto — APPROVED** [review]
LGTM +1

**Pull request merged**
